docs/source/user_guide/data_checks.ipynb

```diff
@@ -454,6 +455,41 @@
     "    print(\"Error:\", message['message'])"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Datetime Format\n",
+    "\n",
+    "Datetime information is a necessary component of time series problems, but sometimes the data we deal with may contain flaws that make it impossible for time series models to work with them. For example, in order to identify a frequency in the datetime information there has to be equal interval spacing between data points i.e. January 1, 2021, January 2, 2021, January 3, 2021, ...etc. If instead there are random jumps in the datetime data i.e. January 1, 2021, January 2, 2021, January 5, 2021, then a frequency can't be inferred. Another common issue with datetime information is if it isn't properly sorted. Datetime values that aren't monotonically increasing or decreasing will encounter the same problem.\n",
+    "\n",
+    "To make it easy to verify that the datetime column you're working with is properly spaced and sorted, we can leverage the `DatetimeFormatDataCheck`. When initializing the data check, pass in the name of the column that contains your datetime information (or pass in \"index\" if it's found in either your X or y indices)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from evalml.data_checks import DateTimeFormatDataCheck\n",
+    "\n",
+    "X = pd.DataFrame(pd.date_range(\"January 1, 2021\", periods=8), columns=[\"dates\"])\n",
+    "y = pd.Series([1, 2, 4, 2, 1, 2, 3, 1])\n",
+    "\n",
+    "# Replaces the last entry with January 9th instead of January 8th\n",
+    "X.iloc[7] = \"January 9, 2021\"\n",
+    "\n",
+    "datetime_format_check = DateTimeFormatDataCheck(datetime_column=\"dates\")\n",
+    "results = datetime_format_check.validate(X, y)\n",
```

Comment:
nit-pick: Might be nice to include the not-monotonic-increasing example as well.